xc/device.ml

```diff
@@ -1536,7 +1552,22 @@ module Vusb = struct
              raise Device_not_connected
       )
       (fun () ->
-         usb_reset_detach ~hostbus ~hostport ~domid ~privileged
+         let usb_path = Printf.sprintf "/local/domain/%d/usb/%s" domid
+             (Stdext.Xstringext.String.split '.' id |> String.concat "-") in
+         let uid_path = Printf.sprintf "%s/%s" usb_path "uid" in
+         let gid_path = Printf.sprintf "%s/%s" usb_path "gid" in
+         let uid =
```

Comment:
This should be one one local function `read` that receives that path such that you can write:
```
let uid = read uid_path in
let gid = read gid_path in
```

Reply:
Ok. Thanks.